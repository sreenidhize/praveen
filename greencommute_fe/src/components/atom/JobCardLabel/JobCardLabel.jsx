import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles({
  primary: {
    width: "250px",
    fontSize: "18px",
    color: "#19293b",
    marginBottom: "8px",
    whiteSpace: "nowrap",
    overflow: "hidden !important",
    textOverflow: "ellipsis",
  },
  secondary: {
    maxWidth: "250px",
    fontSize: "14px",
    color: "#5f7381",
    marginBottom: "8px",
    whiteSpace: "nowrap",
    overflow: "hidden !important",
    textOverflow: "ellipsis",
  },
});

const JobCardLabel = ({ label, style, ...rest }) => {
  const classes = useStyles();
  let textstyle = style ? classes.primary : classes.secondary;
  return (
    <Typography className={textstyle} variant="h4" {...rest}>
      {label}
    </Typography>
  );
};

JobCardLabel.propTypes = {
  label: PropTypes.string,
  style: PropTypes.bool,
};

export default JobCardLabel;
