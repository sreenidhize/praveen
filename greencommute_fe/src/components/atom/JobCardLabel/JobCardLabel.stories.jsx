import React from "react";
import JobCardLabel from "./JobCardLabel";

export default {
  title: "Atoms/CardLabel",
  component: JobCardLabel,
};
const JobCardLabelTemplate = (args) => <JobCardLabel {...args} />;

export const Primary = JobCardLabelTemplate.bind({});

Primary.args = {
  label: "User Experience Designer",
  style: true,
};

export const Secondary = JobCardLabelTemplate.bind({});

Secondary.args = {
  label: "Hi-tech City, Telangana",
  style: false,
};
