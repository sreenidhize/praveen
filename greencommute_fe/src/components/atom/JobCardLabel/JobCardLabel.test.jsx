import React from "react";
import JobCardLabel from "./JobCardLabel";
import { render } from "@testing-library/react";

describe("JobCardLabel renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<JobCardLabel />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("JobCardLabel should not be empty", () => {
    const { container } = render(
      <JobCardLabel label="Hi-tech City, Telangana" style={true} />
    );

    expect(container.getElementsByTagName("h4").length).toBe(1);
  });
});
