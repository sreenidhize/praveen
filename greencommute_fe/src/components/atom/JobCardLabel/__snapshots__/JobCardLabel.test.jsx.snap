// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardLabel renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root makeStyles-secondary-2 MuiTypography-h4"
  />
</DocumentFragment>
`;
